frontend/src/pages/EditProductPage.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Formik, Form } from 'formik'
+import { useParams, useHistory } from 'react-router-dom'
+import { Button } from 'react-bootstrap'
+import InputField from '../components/InputField'
+import SelectCategory from '../components/listofproducts/SelectCategory'
+import validationEdit from '../communications/validationEdit'
+import { updateProductData, fetchOneProduct } from '../communications/userApi'
+import EditResponseModal from '../components/modals/EditResponseModal'
+
+const updateOldFieldsInItem = (newProductValues, product) => {
+  Object.keys(product).forEach((key) => {
+    return newProductValues[key]
+      ? (product[key] = newProductValues[key])
+      : product[key]
+  })
+  return product
+}
+
+const EditProductPage = () => {
+  const [product, setProduct] = useState('')
+  const [showResponseModal, setShowResponseModal] = useState(false)
+  const [responseModalTitle, setResponseModalTitle] = useState('')
+  const [isEditSuccessful, setEditSuccessful] = useState(false)
+  const { productId } = useParams()
+  const { t } = useTranslation()
+  let history = useHistory()
+
+  // These can be later organized into a useProducts hook when product list is ready
+  useEffect(() => {
+    const getProduct = async () => {
+      const response = await fetchOneProduct(productId)
+      setProduct(response.data)
+    }
+    getProduct()
+  }, [productId])
+
+  const handleSubmit = async (newProductValues) => {
+    const updatedProduct = updateOldFieldsInItem(newProductValues, product)
+    try {
+      await updateProductData(product.id, updatedProduct)
+      setResponseModalTitle(t('editProduct.successfulEdition'))
+      setShowResponseModal(true)
+      setEditSuccessful(true)
+    } catch (error) {
+      setResponseModalTitle(t('editProduct.unsuccessfulEdition'))
+      setShowResponseModal(true)
+      setEditSuccessful(false)
+    }
+  }
+
+  const onClose = () => {
+    setShowResponseModal(false)
+    if (isEditSuccessful) {
+      history.push('/super-admin/products')
+    }
+  }
+
+  return (
+    <div className="m-5">
+      <Formik
+        initialValues={{
+          address: '',
```

Comment:
Ha ez egy edit, akkot nem kellen minden értéket visszatölteni az eddigi product-ból? Akár így: initialValues={product}

Reply:
Azt nem szerettem volna, mert akkor a formban úgy kellene kezdeni egy adott mező szerkesztését, hogy kitöröljük a régi értéket. Ez nem gond? initialValues={product} nem tölti fel a kezdeti értékekkel :/

Reply:
Hát szerkesztésnél az eddigi adatokat szeretném átírni, nem akarok mindent újra megadni szerintem.

Reply:
Erre válaszul született meg az updateOldFieldsInItem függvény, amely a ki nem töltött mezők esetében a régi értéket küldi vissza BE-re.